src/lib/kit/components/Inputs/MultiOneOf/MultiOneOf.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+
+import {Select} from '@gravity-ui/uikit';
+import _ from 'lodash';
+
+import {
+    Controller,
+    FieldValue,
+    ObjectIndependentInput,
+    ObjectIndependentInputProps,
+    ValidateError,
+} from '../../../../core';
+import {block} from '../../../utils';
+import {GroupIndent} from '../../GroupIndent';
+
+import './MultiOneOf.scss';
+
+const b = block('multi-oneof');
+
+export interface MultiOneOfProps extends ObjectIndependentInputProps {
+    withoutIndent?: boolean;
+}
+
+export const MultiOneOf: React.FC<MultiOneOfProps> = (props) => {
+    const {name, input, spec, Layout, withoutIndent} = props;
+    const {value, onBlur, onChange, onFocus, parentOnUnmount} = input;
+
+    const [valueSelect, setValueSelect] = React.useState(() => (value ? Object.keys(value) : []));
+
+    const handleOpenChange = React.useCallback(
+        (open: boolean) => {
+            if (open) {
+                onFocus();
+            } else {
+                onBlur();
+            }
+        },
+        [onFocus, onBlur],
+    );
+
+    const parentOnChange = React.useCallback(
+        (
+            childName: string,
+            childValue: FieldValue,
+            childErrors?: Record<string, ValidateError>,
+        ) => {
+            const _value = _.set(value || {}, childName.split(`${name}.`).join(''), childValue);
+
+            onChange(_value, childErrors);
```

Comment:
let's take the current value not from the props, you can pass the callback instead of the value to onChnage, which takes the actual value into itself